src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.g.cs

```diff
@@ -0,0 +1,484 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using System;
+using System.Collections.Immutable;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions
+{
+    /// <summary>
+    /// Generated class that returns SupportedSuppressions for all Sonar C# and VB.NET rules
+    /// </summary>
+    internal sealed class SupportedSuppressionsBuilder
+    {
+        private static readonly Lazy<SupportedSuppressionsBuilder> lazy = new Lazy<SupportedSuppressionsBuilder>(() => new SupportedSuppressionsBuilder());
+
+        public static SupportedSuppressionsBuilder Instance => lazy.Value;
```

Comment:
Singleton pattern, following example (4) in Mr Skeet's blog: https://csharpindepth.com/articles/singleton